tensorrt_llm/_torch/pyexecutor/resource_manager.py

```
@@ -1204,10 +1241,26 @@ def add_dummy_requests(
 
+    def _update_cross_kv_resources(self,
+                                   scheduled_batch: ScheduledRequests) -> None:
+        """Persist cross-attention KV blocks after a scheduled batch.
+        """
+        for request in scheduled_batch.context_requests:
+            self.impl.store_context_blocks(request)
+
     def update_resources(self,
                          scheduled_batch: ScheduledRequests,
                          attn_metadata: "AttentionMetadata" = None,
                          kv_cache_dtype_byte_size: float = None):
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)